networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -335,13 +335,16 @@ def _get_vrf_name(self, ri):
         # that isn't there, look for the router we used to have
         # (indicates that the gateway port is being cleared, but
         # means that the infromation is still useful)
-        ext_gw_port = ri.router.get('gw_port') or ri.ex_gw_port
-        if not ext_gw_port:
-            return None
-        vrf_tag = ext_gw_port['hosting_info'].get('vrf_id')
-        # This could happen if it's the global router
-        if not vrf_tag:
+        if ri.router.get('gw_port') and (
```

Comment:
I guess I'm not convinced your suggestion is an improvement -- the original patch is making a check to see which gateway port object to use. Whichever one it can use, it gets the VRF ID and VLAN/segmentation ID from it.